plugins/schedule/base/services/ScheduleEventService.php

```diff
@@ -316,37 +320,58 @@ private static function setRecurringDates($dates, $dbScheduleEvent)
 
 	/**
 	 * List conflicting events for resourcesIds by event's dates
-	 *
 	 * @action getConflicts
 	 * @param string $resourceIds comma separated
 	 * @param KalturaScheduleEvent $scheduleEvent
 	 * @param string $scheduleEventIdToIgnore
+	 * @param KalturaScheduleEventConflictType $scheduleEventConflictType
 	 * @return KalturaScheduleEventListResponse
+	 * @throws KalturaAPIException
 	 */
-	public function getConflictsAction($resourceIds, KalturaScheduleEvent $scheduleEvent,$scheduleEventIdToIgnore=null)
+	public function getConflictsAction($resourceIds, KalturaScheduleEvent $scheduleEvent, $scheduleEventIdToIgnore = null,
+									   $scheduleEventConflictType = KalturaScheduleEventConflictType::RESOURCE_CONFLICT)
 	{
 		if (!$resourceIds)
+		{
 			throw new KalturaAPIException(KalturaErrors::PROPERTY_VALIDATION_CANNOT_BE_NULL, 'resourceIds');
+		}
 
-		$dbScheduleEvent = $scheduleEvent->toObject();
 		/* @var $dbScheduleEvent ScheduleEvent */
-
+		$dbScheduleEvent = $scheduleEvent->toObject();
 		$events = array();
-
+		$dates = array();
 		if($dbScheduleEvent->getRecurrenceType() === ScheduleEventRecurrenceType::RECURRING)
 		{
 			$maxRecurrences = SchedulePlugin::getScheduleEventmaxRecurrences();
 			$datesGenerator = new DatesGenerator($maxRecurrences, $dbScheduleEvent->getRecurrence()->asArray(), array('KalturaLog', 'debug'));
 			$dates = $datesGenerator->getDates($dbScheduleEvent->getStartDate(null));
-
-			foreach($dates as $date)
-				$events = array_merge($events, ScheduleEventPeer::retrieveEventsByResourceIdsAndDateWindow($resourceIds, $date, ($date + $dbScheduleEvent->getDuration()),$scheduleEventIdToIgnore));
+			$duration = $dbScheduleEvent->getDuration();
 		}
-		else {
-			$events = ScheduleEventPeer::retrieveEventsByResourceIdsAndDateWindow($resourceIds, $dbScheduleEvent->getStartDate(null), $dbScheduleEvent->getEndDate(null),$scheduleEventIdToIgnore);
+		else
+		{
+			$dates[] = $dbScheduleEvent->getStartDate(null);
+			$duration = $dbScheduleEvent->getEndDate(null) - $dbScheduleEvent->getStartDate(null);
 		}
+
+		foreach($dates as $date)
+		{
+			if($scheduleEventConflictType != KalturaScheduleEventConflictType::BLACKOUT_CONFLICT)
+			{
+				$events = array_merge($events, ScheduleEventPeer::retrieveEventsByResourceIdsAndDateWindow($resourceIds,
+					$date, $date + $duration, $scheduleEventIdToIgnore));
+			}
+
+			if($scheduleEventConflictType != KalturaScheduleEventConflictType::RESOURCE_CONFLICT)
+			{
+				$events = array_merge($events, ScheduleEventPeer::retrieveBlackoutEventsByDateWindow($date,

```

Comment:
Since this code is almost duplicated , better add function ScheduleEventPeer::getConflicts($dateStart , $dateEnd, $idToIgnore,$conflictType)